src/vs/workbench/services/textfile/browser/browserTextFileService.ts

```diff
@@ -54,7 +54,7 @@ export class BrowserTextFileService extends AbstractTextFileService {
 	private registerListeners(): void {
 
 		// Lifecycle
-		this.lifecycleService.onBeforeShutdown(event => event.veto(this.onBeforeShutdown(), 'veto.textFiles'));
+		this._register(this.lifecycleService.onBeforeShutdown(event => event.veto(this.onBeforeShutdown(), 'veto.textFiles')));
```

Comment:
A reason for not having registered this listener is that the handler must always trigger to prevent data loss. I think here we are probably save because the `BrowserTextFileService` is not getting disposed, but please verify.